be/src/Unic.UrlMapper2/code/Services/RedirectSearcher.cs

```diff
@@ -83,7 +83,8 @@ protected virtual void HandlePreserveQueryString(Redirect redirect, RedirectSear
             if (!redirect.PreserveQueryString) return;
 
             redirect.RegexEnabled = true;
-            if (!redirectSearchResultItem.SourceTerm.EndsWith(Constants.RegularExpressions.QueryStringExpression))
+            if (!string.IsNullOrWhiteSpace(redirectSearchResultItem.SourceTerm) 
```

Comment:
**issue:** you access the `SourceTerm` twice here. As you can never know how expensive this operation is, it's preferred to avoid this.
**suggestion:** assign the `SourceTerm` property to a variable and check against the variable.